src/android/java/io/jxcore/node/BtToRequestSocket.java

```diff
@@ -165,82 +150,89 @@ public void StreamCopyError(StreamCopyingThread who, String error) {
         }
     }
 
-    public int GetLocalHostPort() {
+    private  int GetLocalHostPort() {
         return mHTTPPort;
     }
 
-    public String GetLocalHostAddress() {
-        String ret = "";
-        if(localHostSocket != null && localHostSocket.getInetAddress() != null){
-            ret = localHostSocket.getInetAddress().toString();
-        }
-        return ret;
+    private String GetLocalHostAddressAsString() {
+        return localHostSocket == null || localHostSocket.getInetAddress() == null ? null : localHostSocket.getInetAddress().toString();
     }
 
     public void Stop() {
         mStopped = true;
```

Comment:
Shouldn't you mark this as volatile and check it before stop and run? In other words if this is true when stop is called then clearly stop has been called twice. Similarly if this is true when start is called then start is being called after stop. Just make sure that member variable is volatile or you can end up with some odd race conditions.